### PR TITLE
ONT-591 fix add lastupdateblocknum for check force update

### PR DESCRIPTION
**Pull request by xiemylogos**
fix add lastupdateblocknum for check force update

### CODE REVIEW

File: consensus/vbft/service.go [hunk 1]
```diff
@@ -304,14 +305,15 @@ func (self *Server) getChainConfig() (*vconfig.ChainConfig, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get peersinfo from leveldb: %s", err)
 	}
-	cfg, err := vconfig.GenesisChainConfig(config, peersinfo)
-	if err != nil {
-		return nil, fmt.Errorf("GenesisChainConfig failed: %s", err)
-	}
 	goverview, err := self.chainStore.GetGovernanceView()
```
Comment (Honglei-Cong) [inline consensus/vbft/service.go]: why GetGovernanceView() is a member function of chainstore? 
if has nothing with chainstore, it should be one util function.

File: consensus/vbft/service.go [hunk 2]
```diff
@@ -2001,7 +2026,7 @@ func (self *Server) creategovernaceTransaction(blkNum uint32) *types.Transaction
 
 //checkNeedUpdateChainConfig use blockcount
 func (self *Server) checkNeedUpdateChainConfig(blockNum uint32) bool {
-	if blockNum%self.config.MaxBlockChangeView == 0 {
+	if (blockNum - self.config.LastUpdateBlockNum) == self.config.MaxBlockChangeView {
```
Comment (Honglei-Cong) [inline consensus/vbft/service.go]: (blockNum - self.config.LastUpdateBlockNum) >= self.config.MaxBlockChangeView ?

File: consensus/vbft/msg_builder.go [hunk 3]
```diff
@@ -185,11 +185,19 @@ func (self *Server) constructBlock(blkNum uint32, prevBlkHash common.Uint256, tx
 	txRoot := common.ComputeMerkleRoot(txHash)
 	blockRoot := ledger.DefLedger.GetBlockRootWithNewTxRoot(txRoot)
 
+	preblock, err := self.chainStore.GetBlock(blkNum - 1)
```
Comment (Honglei-Cong) [inline consensus/vbft/msg_builder.go]: remove these changes.

File: consensus/vbft/service.go [hunk 4]
```diff
@@ -304,14 +305,15 @@ func (self *Server) getChainConfig() (*vconfig.ChainConfig, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get peersinfo from leveldb: %s", err)
 	}
-	cfg, err := vconfig.GenesisChainConfig(config, peersinfo)
```
Comment (Honglei-Cong) [inline consensus/vbft/service.go]: self.chainStore.GetVbftConfigInfo() seems not related with chainstore.

change it to util function.

File: consensus/vbft/service.go [hunk 5]
```diff
@@ -2030,6 +2055,16 @@ func (self *Server) validHeight(blkNum uint32) uint32 {
 	return validHeight
 }
 
+func (self *Server) nonSystxs(sysTxs []*types.Transaction) bool {
+	if self.checkNeedUpdateChainConfig(self.currentBlockNum) && len(sysTxs) == 1 {
+		invoke := sysTxs[0].Payload.(*payload.InvokeCode)
+		if bytes.Compare(invoke.Code.Code, gover.COMMIT_DPOS_BYTES) == 0 {
```
Comment (Honglei-Cong) [inline consensus/vbft/service.go]: check if invoke is nil

File: smartcontract/service/native/governance/states.go [hunk 6]
```diff
@@ -679,6 +679,7 @@ func (this *VoteCommitInfo) Deserialize(r io.Reader) error {
 type GovernanceView struct {
 	View       *big.Int `json:"view"`
 	VoteCommit bool     `json:"voteCommit"`
+	TxHash     common.Uint256
```
Comment (Honglei-Cong) [inline smartcontract/service/native/governance/states.go]: gofmt ?

File: consensus/vbft/config/config.go [hunk 7]
```diff
@@ -48,6 +48,7 @@ type ChainConfig struct {
 	Peers                []*PeerConfig `json:"peers"`
 	PosTable             []uint32      `json:"pos_table"`
 	MaxBlockChangeView   uint32        `json:"MaxBlockChangeView"`
+	LastUpdateBlockNum   uint32
```
Comment (Honglei-Cong) [inline consensus/vbft/config/config.go]: why add this?  
I think there already is VbftBlockInfo.LastConfigBlockNum?

**Review by laizy — DISMISSED** [review]

**Review by laizy — APPROVED** [review]
